src/TransitionSystems/transition_system.rs

```diff
@@ -1,256 +1,344 @@
 use crate::DBMLib::dbm::Federation;
+use crate::EdgeEval::constraint_applyer::apply_constraints_to_state;
 use crate::ModelObjects::component::{
-    Channel, Component, DeclarationProvider, Declarations, DecoratedLocation, Location, State,
-    SyncType, Transition,
+    Channel, Component, DeclarationProvider, Declarations, DecoratedLocation, Location,
+    LocationType, State, SyncType, Transition,
 };
 use crate::ModelObjects::max_bounds::MaxBounds;
+use crate::ModelObjects::representations::QueryExpression;
 use crate::System::local_consistency;
 use crate::System::pruning;
 use dyn_clone::{clone_trait_object, DynClone};
 use std::collections::hash_set::HashSet;
+use std::collections::HashMap;
+use std::fmt::{Display, Formatter};
+
+#[derive(Debug, Clone, Eq, Hash, PartialEq)]
+pub enum LocationID {
+    Conjunction(Box<LocationID>, Box<LocationID>),
+    Composition(Box<LocationID>, Box<LocationID>),
+    Quotient(Box<LocationID>, Box<LocationID>),
+    Simple(String),
+}
```

Comment:
I think its time to move location related structs away from this file. You can decide if LocationId and LocationTuple should share a file or not

Reply:
Done, i moved them to separate files as we might as well have smaller files

Reply:
perfect :D